pallets/attestation/src/lib.rs

```diff
@@ -257,58 +292,32 @@ pub mod pallet {
 			);
 
 			// Check for validity of the delegation node if specified.
-			let delegation_record = if let Some(delegation_id) = delegation_id {
-				let delegation = <delegation::DelegationNodes<T>>::get(delegation_id)
-					.ok_or(delegation::Error::<T>::DelegationNotFound)?;
-
-				ensure!(!delegation.details.revoked, Error::<T>::DelegationRevoked);
-
-				ensure!(delegation.details.owner == who, Error::<T>::NotDelegatedToAttester);
-
-				ensure!(
-					(delegation.details.permissions & delegation::Permissions::ATTEST)
-						== delegation::Permissions::ATTEST,
-					Error::<T>::DelegationUnauthorizedToAttest
-				);
-
-				// Check if the CType of the delegation is matching the CType of the attestation
-				let root = <delegation::DelegationHierarchies<T>>::get(delegation.hierarchy_root_id)
-					.ok_or(delegation::Error::<T>::HierarchyNotFound)?;
-				ensure!(root.ctype_hash == ctype_hash, Error::<T>::CTypeMismatch);
-
-				// If the attestation is based on a delegation, store separately
-				let mut delegated_attestations = <DelegatedAttestations<T>>::get(delegation_id).unwrap_or_default();
-				delegated_attestations
-					.try_push(claim_hash)
-					.map_err(|_| Error::<T>::MaxDelegatedAttestationsExceeded)?;
-				Some((delegation_id, delegated_attestations))
-			} else {
-				None
-			};
+			authorization.as_ref().map(|ac| ac.can_attest(&who)).transpose()?;
+			let authorization_id = authorization.as_ref().map(|ac| ac.authorization_id());
 
 			let deposit = Pallet::<T>::reserve_deposit(payer, deposit_amount)?;
 
 			// *** No Fail beyond this point ***
 
 			log::debug!("insert Attestation");
 
-			// write delegation record, if any
-			if let Some((id, delegated_attestation)) = delegation_record {
-				<DelegatedAttestations<T>>::insert(id, delegated_attestation);
-			}
-
-			<Attestations<T>>::insert(
+			Self::deposit_event(Event::AttestationCreated(
```

Comment:
Good point, I think we didn't talk about that yet.
The delegated attestations can't be part of the attestation pallet, since it would then depend on the delegation pallet (even tho not in code and only in logic).
Do we need this storage? I guess yes. What if a delegation node gets deleted, but there are still attestations remaining. They cant get removed anymore?